File: algorithms/1_easy/S/03550_smallest_index_with_digit_sum_equal_to_index/solution.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        for (index, val) in enumerate(nums):
            dig_sum = 0

            while val > 0 :
                dig_sum += val%10
                val //=10
            if index == dig_sum : 
                return index

        return -1

File: algorithms/1_easy/S/03550_smallest_index_with_digit_sum_equal_to_index/solution.rs
impl Solution {
    pub fn smallest_index(nums: Vec<i32>) -> i32 {
        for (index, mut val) in nums.into_iter().enumerate() {
            let mut dig_sum = 0_i32;

            while val > 0 {
                dig_sum += val % 10;
                val /= 10; 
            }

            if dig_sum == index as i32 {
                return index as i32;
            }
        }

        -1
    }
}

